src/main/java/dev/openfeature/javasdk/Structure.java

```diff
@@ -0,0 +1,87 @@
+package dev.openfeature.javasdk;
+
+import java.time.ZonedDateTime;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+import lombok.EqualsAndHashCode;
+import lombok.ToString;
+
+@ToString
+@EqualsAndHashCode
+@SuppressWarnings("PMD.BeanMembersShouldSerialize")
+public class Structure {
+
+    protected final Map<String, Value> attributes;
+
+    public Structure() {
+        this.attributes = new HashMap<>();
+    }
+
+    public Structure(Map<String, Value> attributes) {
+        this.attributes = new HashMap<>(attributes);
+    }
+
+    public Set<String> keySet() {
+        return this.attributes.keySet();
+    }
+
+    // getters
+    public Value getValue(String key) {
+        return this.attributes.get(key);
+    }
+
+    // adders
+    public Structure add(String key, Boolean value) {
+        attributes.put(key, new Value(value));
+        return this;
+    }
+
+    public Structure add(String key, String value) {
+        attributes.put(key, new Value(value));
+        return this;
+    }
+
+    public Structure add(String key, Integer value) {
+        attributes.put(key, new Value(value));
+        return this;
+    }
+
+    public Structure add(String key, Double value) {
+        attributes.put(key, new Value(value));
+        return this;
+    }
+
+    /** 
+     * Add date-time relevant key.
+     * 
+     * @param key feature key
+     * @param value date-time value
+     * @return Structure
+     */
+    public Structure add(String key, ZonedDateTime value) {
+        attributes.put(key, new Value(value));
+        return this;
+    }
+
+    public Structure add(String key, Structure value) {
+        attributes.put(key, new Value(value));
+        return this;
+    }
+
+    public <T> Structure add(String key, List<Value> value) {
+        attributes.put(key, new Value(value));
+        return this;
+    }
+
+    /**
+     * Get all values.
+     * 
+     * @return all attributes on the structure
+     */
+    public Map<String, Value> asMap() {
+        return new HashMap<>(this.attributes);
```

Comment:
Are we OK with a shallow copy of the values? The application developer will use this to obtain all the values hence a  shallow copy is OK in that scenario. But will this structure also be used during building and mutating the EvaluationContext?

Reply:
I think this is a good question...

I suspect that for mutating, users will be using the `merge` and `add` functions in the EvaluationContext. I don't have any strong opinions about if this should be a shallow copy, deep copy, or just expose the inner map (though I think in most cases, that's a bad choice). I guess whatever we choose it should be well-documented.

Does anyone have a strong argument for this?

@justinabrahms @kinyoklion

Reply:
https://github.com/open-feature/java-sdk/issues/53 Moved this discussion here.